target/linux/ath79/image/generic-tp-link.mk

```diff
@@ -564,6 +564,17 @@ define Device/tplink_tl-wpa8630p-v2-int
 endef
 TARGET_DEVICES += tplink_tl-wpa8630p-v2-int
 
+define Device/tplink_tl-wpa8730-v1
+  $(Device/tplink-8mlzma)
+  SOC := qca9558
+  DEVICE_MODEL := TL-WPA8730
+  DEVICE_VARIANT := v1
+  IMAGE_SIZE := 8064k
+  TPLINK_BOARD_ID := TLWPA8730V1
```

Comment:
TPLINK_BOARD_ID is for safeloader, if you use 8mlzma you need TPLINK_HWID

Reply:
I've removed it, using the default hwid of 0x0, as that's what in the official image:

Raw dump of the original image on the SPI flash:
```
65536         0x10000         TP-Link firmware header, firmware version: 0.0.0, image version: "", product ID: 0x0, product version: 0, kernel load address: 0x0, kernel entry point: 0x80002000, kernel offset: 0, kernel length: 512, rootfs offset: 985512, rootfs length: 0, bootloader offset: 0, bootloader length: 0
66048         0x10200         LZMA compressed data, properties: 0x5D, dictionary size: 8388608 bytes, uncompressed size: 2847232 bytes
1114112       0x110000        Squashfs filesystem, little endian, version 4.0, compression:lzma, size: 4043374 bytes, 961 inodes, blocksize: 65536 bytes, created: 2017-04-07 04:15:58
```

Matching firmware update image `wpa8730v1_1_0_2_up(170407)-20170407.bin`:
```
56            0x38            TP-Link firmware header, firmware version: 0.0.0, image version: "", product ID: 0x0, product version: -2026897407, kernel load address: 0x0, kernel entry point: 0x0, kernel offset: 0, kernel length: 512, rootfs offset: 1048576, rootfs length: 1049088, bootloader offset: 7142400, bootloader length: 0
568           0x238           TP-Link firmware header, firmware version: 0.0.0, image version: "", product ID: 0x0, product version: 0, kernel load address: 0x0, kernel entry point: 0x80002000, kernel offset: 0, kernel length: 512, rootfs offset: 985512, rootfs length: 0, bootloader offset: 0, bootloader length: 0
1080          0x438           LZMA compressed data, properties: 0x5D, dictionary size: 8388608 bytes, uncompressed size: 2847232 bytes
1049144       0x100238        Squashfs filesystem, little endian, version 4.0, compression:lzma, size: 4043374 bytes, 961 inodes, blocksize: 65536 bytes, created: 2017-04-07 04:15:58
```